src/app/ecosystem-explorer/utils/createFeaturedProjectsZodField.ts

```diff
@@ -0,0 +1,20 @@
+import { z } from 'zod'
+
+import { extractSlugFromFilename } from '@/utils/fileUtils'
+
+import { MarkdownPathSchema } from '@/schemas/FrontmatterSchema'
+
+import { getEcosystemProjectsData } from './getEcosystemProjectData'
+
+export function createFeaturedProjectsZodField(errorMessage: string) {
+  return z
+    .array(MarkdownPathSchema)
+    .transform((paths) => {
+      const ecosystemProjects = getEcosystemProjectsData()
+      const slugs = paths.map(extractSlugFromFilename)
+      return ecosystemProjects.filter((item) => slugs.includes(item.slug))
+    })
+    .refine((filtered) => filtered.length > 0, {
+      message: errorMessage,
+    })
```

Comment:
Think I might be missing something - what is the case when we would see the custom error message?

If the path is incorrect or file is missing, it would be caught by `MarkdownPathSchema`?

Reply:
> Think I might be missing something - what is the case when we would see the custom error message?

Empty array..

Reply:
Ah I commented out and it left an ","  in the array so it didn't show the right message. Sorry!